mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/modules/CassandraMessageModule.java

```diff
@@ -47,12 +47,26 @@
     private final List<CassandraType> types;
 
     public CassandraMessageModule() {
-        tables = Collections.singletonList(
-            new CassandraTable(CassandraMessageTable.TABLE_NAME,
-                SchemaBuilder.createTable(CassandraMessageTable.TABLE_NAME)
+        tables = ImmutableList.of(
+            new CassandraTable(CassandraMessageTable.MESSAGE_ID_TABLE,
+                SchemaBuilder.createTable(CassandraMessageTable.MESSAGE_ID_TABLE)
                     .ifNotExists()
                     .addPartitionKey(CassandraMessageTable.MAILBOX_ID, timeuuid())
                     .addClusteringColumn(CassandraMessageTable.IMAP_UID, bigint())
+                    .addColumn(CassandraMessageTable.MESSAGE_ID, text())),
+            new CassandraTable(CassandraMessageTable.IMAP_UID_TABLE,
+                    SchemaBuilder.createTable(CassandraMessageTable.IMAP_UID_TABLE)
+                        .ifNotExists()
+                        .addPartitionKey(CassandraMessageTable.MESSAGE_ID, text())
+                        .addClusteringColumn(CassandraMessageTable.MAILBOX_ID, timeuuid())
+                        .addColumn(CassandraMessageTable.IMAP_UID, bigint())),
+
+            new CassandraTable(CassandraMessageTable.TABLE_NAME,
+                SchemaBuilder.createTable(CassandraMessageTable.TABLE_NAME)
+                    .ifNotExists()
+                    .addPartitionKey(CassandraMessageTable.MESSAGE_ID, text())
+                    .addColumn(CassandraMessageTable.MAILBOX_ID, timeuuid())
```

Comment:
I have to remove those two columns(MAILBOX_ID, IMAP_UID), the indices should provide it.